arch/common/isr_tables.c

```diff
@@ -21,7 +21,7 @@ struct int_list_header {
  * header of the initList section, which is used by gen_isr_tables.py to create
  * the vector and sw isr tables,
  */
-Z_GENERIC_SECTION(.irq_info) struct int_list_header _iheader = {
+Z_GENERIC_SECTION(.irq_info) __used struct int_list_header _iheader = {
```

Comment:
We had an issue while enabling LTO during tests done by @LuDuda. The header was optimized out. This fix removes the issue.

Reply:
I know that using LTO can cause some surprising problems. In a matter of fact, with LTO, linker does some jobs that are more compiler related. Now it is linker job to prove that some functions, structures and others are not referenced. Note also, that we are adding -fffunction-sections and -fdata-sections - maybe this would help to remove this variable if unused. It surprises me a little as it is placed in predefined section anyway. But it is the effect we had and this was the fastest and simplest solution.